definitions/tools/cellranger_count.cwl

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "Run Cell Ranger Count"
+
+baseCommand: ["cellranger", "count", "--localmem=64", "--localcores=8"]
+
+requirements:
+    - class: DockerRequirement
+      dockerPull: "registry.gsc.wustl.edu/ebelter/cellranger:2.2.0"
+    - class: ResourceRequirement
+      ramMin: 64000
+      coresMin: 8
+
+inputs:
+    chemistry:
+        type: string?
+        inputBinding:
+            prefix: --chemistry=
+            position: 1
+            separate: false
+        default: "auto"
+        doc: "Assay configuration used, default 'auto' should usually work without issue"
+    fastq_directory:
+        type: Directory
+        inputBinding:
+           prefix: --fastqs=
+           position: 2
+           separate: false
+        doc: "Directory containing fastq files"
+    reference:
+        type: string
+        inputBinding:
+            prefix: --transcriptome=
+            position: 3
+            separate: false
+        doc: "Transcriptome reference compatible with input species and Cell Ranger"
+    sample_name:
+        type: string
+        inputBinding:
+            prefix: --sample=
+            position: 4
+            separate: false
+        doc: "Sample name, must be same as name specified in sample sheet in previous mkfastq step"
+    unique_id:
```

Comment:
As is (if I'm parsing this right), we'll always have to glob for `workflow_output/UNIQUEID/...` in the worfklow dir. Since workflows all get their own output directory anyway, is there any reason not to just hardcode this to be "cellranger_count" or something like that?  Let me know if I'm missing something...

Reply:
Yes we could hard code it. I just liked having that option to allow you to easily know what's in the directory. What I've been doing is just adding the name of the tool to the end of the sample name. "sample1_count" or "sample1_vdj" and I realize now we can use the sample name input to create the unique id and remove the need for this in the yaml

Reply:
I'm just thinking in terms of consistency across pipelines too.  The alignment, RNAseq, etc workflows don't have subdirs with sample names. They just contain the outputs.  That makes some operations easier: If I have a list of output dirs, it's easy to grab all of some result file without extra globbing.  We probably should get some feedback from the others and go with whatever the consensus is!

Reply:
required param, but could be set to something like "cellranger_output" so that it's consistent across samples.  We actually need to retain the whole directory and substructure, but it can be named whatever.